Adwaita/adwaita.css

```diff
@@ -0,0 +1,166 @@
+/*
+thank you adwaita <333
+
+https://gitlab.gnome.org/GNOME/gtk/-/blob/gtk-3-24/gtk/theme/Adwaita/gtk-contained-dark.css
+*/
+
+body.adwaita {
+    /*
+    * custom variables to be more expressive about what the colors actually set.
+    * (not that heroic folks did a bad job naming these!)
+    */
+    --text: #BFC3C4;
+    --custom-accent: #15539E;
+    --custom-accent-hover: #225a9e; 
+    --background: #232323;
+    --secondary-background: #353535;
+
+    --button-background: #2d2d2d;
+    --button-text: white;
+
+    --box-shadow-color: #1b1b1b;
+    --secondary-box-shadow-color: #202020;
+    --tertiary-box-shadow-color: #5a5a59;
+
+    --box-shadow: 0 0 0 1px var(--box-shadow-color); 
+    --secondary-box-shadow: var(--box-shadow-color);
+
+    --border-radius: var(--space-xs);
+
+
+    --accent: var(--text);
+    /*The text in the navbar, of the selected element*/
+    --accent-overlay: #fefefe;
+    /*The text in the navbar, which isn't selected*/
+    --navbar-accent: var(--text);
+    
+    /*the color of the divider*/
+    --divider: var(--secondary-box-shadow-color);
+
+    --body-background: var(--background);
+    --background-darker: var(--background);
+    --current-background: var(--background);
+    --navbar-background: var(--secondary-background);
+    --navbar-active-background: var(--custom-accent);
+    --gradient-body-background: var(--background);
+    --input-background: var(--secondary-background);
+    --modal-background: var(--background);
+    --modal-border: var(--box-shadow-color);
+    --download-button: #44475a;
+    --download-button-overlay: #007bff;
+    --success: var(--custom-accent);
+    --success-hover: var(--custom-accent-hover);
+    --primary: var(--custom-accent);
+    --primary-hover: var(--custom-accent-hover);
+    --danger: #ff0000;
+    --text-title: #00ddff;
+    --icons-background: var(--secondary-background);
+    --osk-background: var(--background);
+    --osk-button-background: var(--secondary-background);
+    --osk-button-border: #ffffffff;
+  }
+
+body.adwaita .Field>input, 
+body.adwaita select, 
+body.adwaita .gameCard, 
+body.adwaita .gameCard>span,
+body.adwaita .SearchBar,
+body.adwaita .progressHeader,
+body.adwaita .dmItemList
```

Comment:
Please try to use just one `body.adwaita` selector if possible. The other selectors can be placed inside the parent class instead of outside.
```css
body.adwaita {
  gameCard {
  ...
  }
}

Reply:
As a reference - https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/main/src/frontend/themes.scss#L325

Reply:
Unfortunately neither vs code nor web storm support nesting of css selectors.

Also only chrome does:
https://caniuse.com/css-nesting
https://stackoverflow.com/questions/4564916/nesting-css-classes

Do you know how to add support of this to vs code?

Reply:
@redromnon 

I looked deeper into it and the support is VERY bad. Chromium should support it => It should work in Electron.

But I don't think having such code that can't be edited by any IDE in the codebase is a really bad practice, and if it is ok with you, I would rather avoid it.

I could use a preprocessor like sass though :3
Those support nesting

Reply:
No it does not. This shouldn't even work.
Let me elaborate:

The functionality you wan't should be exact this one:
```css
body.adwaita {
    --text: #BFC3C4;
    --custom-accent: #15539E;
    .
    .
    .
    --osk-button-background: var(--secondary-background);
    --osk-button-border: #ffffffff;
 }
    /*Then use the selectors here*/
  body.adwaita  .Field>input, 
body.adwaita  select,  
body.adwaita .gameCard,
body.adwaita .gameCard>span, 
body.adwaita .SearchBar, 
body.adwaita .progressHeader, 
body.adwaita .dmItemList{
      box-shadow: var(--box-shadow) !important;
 }
    
body.adwaita .store-icon {
      background: none;
    }
    
body.adwaita .Sidebar {
      box-shadow: 1px 0 0 0 var(--box-shadow-color);
    }
    /*rest of the code*/
```

You wanna do this nested.
According to the css proposal and due to chromium adding it the syntax for nested things which translates to the same functionality would be:

```css
body.adwaita {
    --text: #BFC3C4;
    --custom-accent: #15539E;
    .
    .
    .
    --osk-button-background: var(--secondary-background);
    --osk-button-border: #ffffffff;
    
    /*Then use the selectors here*/
    & .Field>input, 
    & select,  
    & .gameCard,
    & .gameCard>span, 
    & .SearchBar, 
    & .progressHeader, 
    & .dmItemList{
      box-shadow: var(--box-shadow) !important;
    }
    
    & .store-icon {
      background: none;
    }
    
    & .Sidebar {
      box-shadow: 1px 0 0 0 var(--box-shadow-color);
    }
    /*rest of the code*/
}
```

where `&` represents `body.adwaita`

but again, even this is very badly supported

Reply:
I didn't realize Heroic uses Sass for the theming code, so it's a mistake on my part. I understand now. 

The Heroic devs will anyway port your CSS code to Sass for the upcoming releases.